tool/testdata/wal_dump

```diff
@@ -3,106 +3,106 @@ wal dump
 requires at least 1 arg(s), only received 0
 
 wal dump
-../testdata/db-stage-2/000003.log
+../testdata/db-stage-2/000002.log
 ----
-000003.log
-0(21) seq=1 count=1
+000002.log
+0(21) seq=10 count=1
     SET(test formatter: foo,test value formatter: one)
-28(21) seq=2 count=1
+32(21) seq=11 count=1
     SET(test formatter: bar,test value formatter: two)
-56(23) seq=3 count=1
+64(23) seq=12 count=1
     SET(test formatter: baz,test value formatter: three)
-86(22) seq=4 count=1
+98(22) seq=13 count=1
     SET(test formatter: foo,test value formatter: four)
-115(17) seq=5 count=1
+131(17) seq=14 count=1
     DEL(test formatter: bar)
 EOF
 
 wal dump
-../testdata/db-stage-2/000003.log
+../testdata/db-stage-2/000002.log
 --key=pretty:leveldb.BytewiseComparator
 --value=size
 ----
-000003.log
-0(21) seq=1 count=1
+000002.log
+0(21) seq=10 count=1
     SET(foo,<3>)
-28(21) seq=2 count=1
+32(21) seq=11 count=1
     SET(bar,<3>)
-56(23) seq=3 count=1
+64(23) seq=12 count=1
     SET(baz,<5>)
-86(22) seq=4 count=1
+98(22) seq=13 count=1
     SET(foo,<4>)
-115(17) seq=5 count=1
+131(17) seq=14 count=1
     DEL(bar)
 EOF
 
 wal dump
 --key=pretty:leveldb.BytewiseComparator
 --value=size
-../testdata/db-stage-4/000006.log
+../testdata/db-stage-4/000005.log
 ----
-000006.log
-0(22) seq=6 count=1
+000005.log
+0(22) seq=15 count=1
     SET(foo,<4>)
-29(22) seq=7 count=1
+33(22) seq=16 count=1
     SET(quux,<3>)
-58(17) seq=8 count=1
+66(17) seq=17 count=1
     DEL(baz)
 EOF
 
 wal dump
-../testdata/db-stage-4/000006.log
+../testdata/db-stage-4/000005.log
 --key=%x
 --value=%x
 ----
-000006.log
-0(22) seq=6 count=1
+000005.log
+0(22) seq=15 count=1
     SET(666f6f,66697665)
-29(22) seq=7 count=1
+33(22) seq=16 count=1
     SET(71757578,736978)
-58(17) seq=8 count=1
+66(17) seq=17 count=1
     DEL(62617a)
 EOF
 
 wal dump
-../testdata/db-stage-4/000006.log
+../testdata/db-stage-4/000005.log
 --key=pretty:leveldb.BytewiseComparator
 --value=pretty:test-comparer
 ----
-000006.log
-0(22) seq=6 count=1
+000005.log
+0(22) seq=15 count=1
     SET(foo,test value formatter: five)
-29(22) seq=7 count=1
+33(22) seq=16 count=1
     SET(quux,test value formatter: six)
-58(17) seq=8 count=1
+66(17) seq=17 count=1
     DEL(baz)
 EOF
 
 wal dump
-../testdata/db-stage-4/000006.log
+../testdata/db-stage-4/000005.log
 --key=pretty:test-comparer
 --value=%x
 ----
-000006.log
-0(22) seq=6 count=1
+000005.log
+0(22) seq=15 count=1
     SET(test formatter: foo,66697665)
-29(22) seq=7 count=1
+33(22) seq=16 count=1
```

Comment:
Most of this seems to be due to our sequence number reservation work for disagg storage, which is now unnecessary. We can consider undoing it if we think that's valuable.